contracts/0.8.9/WithdrawalVault.sol

```diff
@@ -122,4 +141,66 @@ contract WithdrawalVault is Versioned {
 
         _token.transferFrom(address(this), TREASURY, _tokenId);
     }
+
+    /**
+     * @dev Submits EIP-7002 full withdrawal requests for the specified public keys.
+     *      Each request instructs a validator to fully withdraw its stake and exit its duties as a validator.
+     *      Refunds any excess fee to the caller after deducting the total fees,
+     *      which are calculated based on the number of public keys and the current minimum fee per withdrawal request.
+     *
+     * @param pubkeys A tightly packed array of 48-byte public keys corresponding to validators requesting full withdrawals.
+     *                | ----- public key (48 bytes) ----- || ----- public key (48 bytes) ----- | ...
+     *
+     * @notice Reverts if:
+     *         - The caller does not have the `ADD_FULL_WITHDRAWAL_REQUEST_ROLE`.
+     *         - Validation of any of the provided public keys fails.
+     *         - The provided total withdrawal fee is insufficient to cover all requests.
+     *         - Refund of the excess fee fails.
+     */
+    function addFullWithdrawalRequests(
+        bytes calldata pubkeys
+    ) external payable onlyRole(ADD_FULL_WITHDRAWAL_REQUEST_ROLE) {
+        uint256 prevBalance = address(this).balance - msg.value;
+
+        uint256 minFeePerRequest = TriggerableWithdrawals.getWithdrawalRequestFee();
+        uint256 totalFee = (pubkeys.length / TriggerableWithdrawals.PUBLIC_KEY_LENGTH) * minFeePerRequest;
```

Comment:
Can we make it a separate view func so that integrating contracts can re-use it?